### ISSUE TITLE
We need to display a trail that shows the travelled path of the drone. This can be a thin perforated line drawn between every position update. This will help users identify the path that drone has taken so far in the video.

### ISSUE THREAD

**Issue by pranjalbansal787** [openlayers/openlayers#14735]
We need to display a trail that shows the travelled path of the drone. This can be a thin perforated line drawn between every position update. This will help users identify the path that drone has taken so far in the video.

Below is the existing codebase . Edit the following code to get the desired result.But this has to be done using open layers only. Help me out.

```js
import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { hasEmptyValue, ToggleSwitch, Text } from 'lib';

import '../../stylesheets/components/Video/VideoMap.scss';

import { dronetelemetry, location } from '../../assets/icons';

import Map from '../common/Map';
import FullScreenControl from '../common/Map/FullscreenControl';
import ZoomControls from '../common/Map/ZoomControls';
import BasemapControl from '../common/Map/BasemapControl';
import MarkersList from '../common/Map/MarkersList';

import coordinatesToPointFeature from '../../utils/map/coordinatesToPointFeature';

const VideoMap = ({ site, telemetry, isPaused }) => {
    const [marker, setMarker] = useState(null);
    const [isAutofocus, toggleAutofocus] = useState(false);
    const ref = useRef(0);

    useEffect(() => {
        let interval;
        if (hasEmptyValue(telemetry)) {
            if (!hasEmptyValue(site)) {
                setMarker({
                    siteType: site.siteType,
                    name: site.name,
                    uuid: site.uuid,
                    geojson: !hasEmptyValue(site.demarcationGeojson)
                        ? site.demarcationGeojson
                        : coordinatesToPointFeature(site.locCoordinates),
                });
            }
        } else if (!isPaused) {
            interval = setInterval(() => {
                const geojson = coordinatesToPointFeature(telemetry[ref.current]);
                setMarker({
                    siteType: site.siteType,
                    name: site.name,
                    uuid: site.uuid,
                    geojson,
                });
                // should be removed once we get an actual file
                ref.current = ref.current === telemetry.length - 1 ? 0 : ref.current + 1;
            }, 300);
        }
        return () => clearInterval(interval);
    }, [telemetry, site, isPaused]);

    if (!hasEmptyValue(site) && !hasEmptyValue(marker)) {
        return (
            <div className="VideoMap">
                <Map>
                    <FullScreenControl />
                    <div>
                        <ToggleSwitch
                            name="setAutofocus"
                            checked={isAutofocus}
                            onChange={() => toggleAutofocus(!isAutofocus)}
                        />
                        <Text text="Enable/Disable Refocus on Map" />
                    </div>
                    <ZoomControls />
                    <BasemapControl />
                    <MarkersList
                        markers={[marker]}
                        textKey="siteType"
                        iconUrl={!hasEmptyValue(telemetry) ? dronetelemetry : location}
                        hoverIconUrl={!hasEmptyValue(telemetry) ? dronetelemetry : location}
                        onClick={() => { }}
                        autofocus={isAutofocus}
                    />
                </Map>
            </div>
        );
    }
    return null;
};

VideoMap.propTypes = {
    site: PropTypes.shape({
        uuid: PropTypes.string,
        name: PropTypes.string,
        siteType: PropTypes.string,
        demarcationGeojson: PropTypes.object,
        locCoordinates: PropTypes.shape({
            lat: PropTypes.number,
            lng: PropTypes.number,
        }),
    }).isRequired,
    telemetry: PropTypes.arrayOf(PropTypes.shape({
        lat: PropTypes.string,
        lng: PropTypes.string,
    })),
    isPaused: PropTypes.bool,
};

VideoMap.defaultProps = {
    telemetry: [],
    isPaused: true,
};

export default VideoMap;
```
<img width="920" alt="image" src="https://user-images.githubusercontent.com/49375731/236975670-faec17a3-b1b0-4395-bf17-74d0d0cd972c.png">

**Comment by mike-000** [comment]
Append the coordinates to a LineString and style that with a line dash stroke.  This is tracking the ISS but the same logic could be used to follow any moving object https://codesandbox.io/s/simple-forked-fhf067?file=/main.js

**Comment by ahocevar** [comment]
@pranjalbansal787 The components you mentioned are not part of this library, but of your application. Please talk to your coworkers that created these components to get help with them. If you're having an issue with OpenLayers, please describe it in detail and provide a minimal example to help reproduce it. Otherwise, please close this issue.

**Comment by tarangrathod** [comment]
do you want something like this 
![image](https://github.com/openlayers/openlayers/assets/72619617/0a07e476-6adc-4e48-bab6-5cd32f0a8372)
i built it with Openlayer itself. if you find good i'll share the code.

**Comment by tarangrathod** [comment]
do you want trail like this !!

will be hapy to share the code build completely with openlayers.
![ezgif-4-6680753f10](https://github.com/openlayers/openlayers/assets/72619617/8deceb5f-8661-4476-b665-1aa7dcc3ce5a)

**Comment by tarangrathod** [comment]
here is my solution to it.
i have live vehicle tracking where we get GPS feeds from mobile application every time there is update in movement of the vehicle and it draws a trace from where it travelled to. so what we do is we establish a Socket connection and that socket connection returns us all the last updated points from the gps table and returns a trail in array format and on UI we plot that into a line string. here the real play is from backend which keeps track of array and sort it by time.
in you code i am not sure what you are doing exactly but here i present the way I do it.

```
   let trailFeatures: any = new GeoJSON().readFeatures(res, { dataProjection: 'EPSG:4326', featureProjection: 'EPSG:3857' })
   trailFeatures = trailFeatures.filter(async (item: any) => {
                    let hasGeometry = item.getGeometry().getCoordinates()
                    let geometryType = item.getGeometry().getType()
                 
                      if (hasGeometry.length && geometryType == 'LineString') {
                        item.getGeometry().forEachSegment(async function (start, end, idx) {
                          let firstpt = item.getGeometry().getFirstCoordinate()
                          styles = (feature: any) => {
                            let driverCd = feature.get('driverCd')
                            let style = [
                              new Style({
                                geometry: new Point(firstpt),
                                image: new Icon({
                                  src: `./truck-top.svg`,
                                  anchor: [0.5, 0.5],
                                  rotateWithView: true,
                                  rotation: feature.get('direction'),
                                  scale: .25,
                                }),
                              }),
                              new Style({
                                stroke: new Stroke({
                                  color: '#00AFB9',
                                  width: 4,
                                }),
                              })
                            ]
                            return style
                          }
                        })
                        item.setStyle(styles)
                        return item
                      }
                      else {
                        styles = (feature: any) => {
                          let style = [
                            new Style({
                              image: new Icon({
                                src: `${./truck-top.svg`,
                                anchor: [0.5, 0.5],
                                scale: .15,
                              }),
                            }),
                            new Style({
                              stroke: new Stroke({
                                color: '#ffcc33',
                                width: 2,
                              }),
                            })
                          ]
                          return style
                        }
                        item.setStyle(styles)
                      }
                  })

                  ``` 
once this is done we plot trailFeatures to map layer that we designated for plotting this trace.
we remove previous draw and draw a new one. which happens instantaneously and it does not makes us feel 
that animation is not real we can also update the same layer with last updated value by getting the difference
 and adding only new ones using appendCoordinate() method in line string.!
Also we do this for more than 1 vehicles we tried for at most 10 vehicles in a single run.
